Scripts/Mac/BuildGVFSForMac.sh

```diff
@@ -56,6 +56,9 @@ dotnet build $VFS_SRCDIR/GVFS.sln --runtime osx-x64 --framework netcoreapp2.1 --
 NATIVEDIR=$VFS_SRCDIR/GVFS/GVFS.Native.Mac
 xcodebuild -configuration $CONFIGURATION -workspace $NATIVEDIR/GVFS.Native.Mac.xcworkspace build -scheme GVFS.Native.Mac -derivedDataPath $VFS_OUTPUTDIR/GVFS.Native.Mac || exit 1
 
+USERNOTIFICATIONDIR=$VFS_SRCDIR/GVFS/GVFS.Notifications/VFSForGit.Mac
+xcodebuild -configuration $CONFIGURATION -project $USERNOTIFICATIONDIR/VFSForGit.xcodeproj build -scheme VFSForGit -derivedDataPath $VFS_OUTPUTDIR/GVFS.Notifications/VFSForGit.Mac || exit 1
```

Comment:
Should this project be built as part of `GVFS.Native.Mac.xcworkspace`?

`GVFS.Native.Mac.xcworkspace` currently includes all of the native Mac VFS4G projects:

```
GVFS/GVFS.PostIndexChangedHook/GVFS.PostIndexChangedHook.Mac.xcodeproj/project.pbxproj
GVFS/GVFS.ReadObjectHook/GVFS.ReadObjectHook.Mac.xcodeproj/project.pbxproj
GVFS/GVFS.VirtualFileSystemHook/GVFS.VirtualFileSystemHook.Mac.xcodeproj/project.pbxproj
```
cc: @nickgra @jrbriggs @jeschu1 @jamill for their thoughts as well

Reply:
We got rid of the original `PrjFS.xcworkspace` we had for the kext code to move to a single .xcodeproj to work around some clunkiness around how workspaces work. I think `GVFS.Native.Mac.xcworkspace` was set up the in the way that it is because we were just copying what we did for the `PrjFS.xcworkspace`, I can't think of a good reason for it to be structured as a workspace.

I think we should convert the existing `GVFS.Native.Mac` workspace to a project (and then include this project as well), although that can be in a separate follow up. 

Adding @pmj as well since had opinions on this in the past.

Reply:
Xcode Workspaces are very loose aggregations of Projects. The only thing you really can do with them is create super-project build Schemes. One of the main downsides is that they don't allow cross-references between projects - e.g. you can't have a target in one project be a build dependency for a target in another project if the only connection between those projects is that they're contained in the same workspace.

You *can* however nest projects - in this case, the outer project can depend on targets from the inner project. This isn't exactly the same as having the targets in the same project of course - project-wide properties aren't propagated to sub-project targets, and the sub-project's targets can't depend on the super-project's targets.

Without looking at the merits of the different approaches for this specific case, my order of preference is normally:

 1. Add new "projects" (apps, tools, libraries, etc.) as *targets* to an existing Xcode Project.
 2. If the new thing really doesn't fit with an existing Xcode Project but uses something from an existing Xcode Project, then nest the existing Project inside a new Project containing the new target. The new target can then e.g. link against libraries built by the existing Project.
 3. If the new thing is only extremely loosely related to the existing Xcode Project, but you want to be able to build everything in one go from within Xcode, create a new, unrelated Project for the new thing, then nest both it and the existing Project inside a newly created umbrella project. The umbrella project contains an aggregate build target which simply lists targets from both sub-projects as dependencies.

I'm not aware of any scenario where you'd prefer a workspace over any of the above.

Reply:
Thanks for the detailed answer Phil, I agree with your assessment that we could convert the workspace into a project. I've opened #1167 to track the conversion.

@alameenshah I think we can leave this as is for now and rework it as part of 1167.